lib/services/role.go

```diff
@@ -1758,6 +1536,90 @@ func (set RoleSet) CheckLoginDuration(ttl time.Duration) ([]string, error) {
 	return out, nil
 }
 
+// CheckAccessToRemoteCluster checks if a role has access to remote cluster. Deny rules are
+// checked first then allow rules. Access to a cluster is determined by
+// namespaces, labels, and logins.
+//
+// Note, logging in this function only happens in debug mode, this is because
+// adding logging to this function (which is called on every server returned
+// by GetNodes) can slow down this function by 50x for large clusters!
+func (set RoleSet) CheckAccessToRemoteCluster(rc RemoteCluster) error {
+	if len(set) == 0 {
+		return trace.AccessDenied("access to cluster denied")
+	}
+
+	var errs []error
+
+	rcLabels := rc.GetMetadata().Labels
+
+	// For backwards compatibility, if there is no role in the set with labels and the cluster
+	// has no labels, assume that the role set has access to the cluster.
+	usesLabels := false
+	for _, role := range set {
+		if len(role.GetClusterLabels(Allow)) != 0 || len(role.GetClusterLabels(Deny)) != 0 {
+			usesLabels = true
+			break
+		}
+	}
+
+	if usesLabels == false && len(rcLabels) == 0 {
```

Comment:
`!usesLabels` otherwise this might be flagged by the linter.

Reply:
we have the linter exception for this clause on